File: minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int getMinimumDifference(TreeNode* root) {
        vector<int> sorted;
        inorder(root, sorted);
        
        int min_diff = sorted.back();
        for (int i = 1; i < sorted.size(); ++i){
            min_diff = min(min_diff, sorted[i] - sorted[i - 1]);
        }
        return min_diff;
    }
private:
    void inorder(TreeNode* node, vector<int>& sorted){
        if (!node) return;
        if (node->left) inorder(node->left, sorted);
        sorted.push_back(node->val);
        if (node->right) inorder(node->right, sorted);
    }
};

File: minimum-absolute-difference-in-bst/minimum-absolute-difference-in-bst.py
import numpy as np
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        res = []
        def in_order(node):
            if not node: return
            if node.left: in_order(node.left)
            res.append(node.val)
            if node.right: in_order(node.right)
        
        in_order(root)
        
        return min(np.abs(np.diff(res)))